mobile_verifier/src/reward_shares.rs

```diff
@@ -203,52 +204,147 @@ pub fn dc_to_mobile_bones(dc_amount: Decimal, mobile_bone_price: Decimal) -> Dec
         .round_dp_with_strategy(DEFAULT_PREC, RoundingStrategy::ToPositiveInfinity)
 }
 
-#[derive(Default)]
-pub struct RadioShares {
-    radio_shares: HashMap<Option<String>, Decimal>,
+#[derive(Debug)]
+struct RadioPoints {
+    heartbeat_multiplier: Decimal,
+    coverage_object: Uuid,
+    seniority: DateTime<Utc>,
+    points: Decimal,
 }
 
-impl RadioShares {
-    fn total_shares(&self) -> Decimal {
-        self.radio_shares
-            .values()
-            .fold(Decimal::ZERO, |sum, amount| sum + amount)
+impl RadioPoints {
+    fn new(heartbeat_multiplier: Decimal, coverage_object: Uuid, seniority: DateTime<Utc>) -> Self {
+        Self {
+            heartbeat_multiplier,
+            seniority,
+            coverage_object,
+            points: Decimal::ZERO,
+        }
+    }
+
+    fn points(&self) -> Decimal {
```

Comment:
it is a lil confusing have the var points in the struct and then this points function.  Difficult to distinguish in their usage and could be error prone where one is used instead of the other.  Maybe rename points function to points_multiplier() or other ?

Reply:
i think it makes sense in this case since the `points` field in the struct is private, so this method is the only way to access that value